### PR TITLE
fix spike_vector_to_indices()

### PR DESCRIPTION
**Pull request by samuelgarcia**
Hi @chrishalcrow
This a better change than #2981 I think becaus the cause was deepp inside.

Can you test if it fixes the sapike amplitude ?

### CODE REVIEW

File: src/spikeinterface/core/sorting_tools.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def spike_vector_to_spike_trains(spike_vector: list[np.array], unit_ids: np.arra
     return spike_trains
 
 
-def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
+def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array, absolut_index=False):
```
Comment (alejoe91) [inline src/spikeinterface/core/sorting_tools.py:50]:
```suggestion
def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array, absolute_index :  bool = False):
```

**Comment by chrishalcrow** [comment]
Yup, this fixes both the `spike_amplitude` and `spike_locations` segment bug.

File: src/spikeinterface/core/sorting_tools.py [hunk 2]
```diff
@@ -61,6 +61,12 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
         List of spike vectors optained with sorting.to_spike_vector(concatenated=False)
     unit_ids: np.array
         Unit ids
+    absolut_index: bool, default False
```
Comment (zm711) [inline src/spikeinterface/core/sorting_tools.py:64]:
```suggestion
    absolute_index: bool, default False
```

File: src/spikeinterface/core/sorting_tools.py [hunk 3]
```diff
@@ -61,6 +61,12 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
...
+        Give spike indices absolut usefull when having a unique spike vector
+        or relative to segment usefull with a list of spike vectors
+        When a unique spike vectors (or amplitudes) is used then absolut_index should be True.
```
Comment (zm711) [inline src/spikeinterface/core/sorting_tools.py:67]:
```suggestion
        When a unique spike vectors (or amplitudes) is used then absolute_index should be True.
```

File: src/spikeinterface/core/sorting_tools.py [hunk 4]
```diff
@@ -61,6 +61,12 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
...
+        When a list of spikes (or amplitudes) is used then absolut_index should be False.
```
Comment (zm711) [inline src/spikeinterface/core/sorting_tools.py:68]:
```suggestion
        When a list of spikes (or amplitudes) is used then absolute_index should be False.
```

File: src/spikeinterface/core/sorting_tools.py [hunk 5]
```diff
@@ -82,12 +88,19 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
 
     num_units = unit_ids.size
     spike_indices = {}
+    
+    total_spikes = 0
     for segment_index, spikes in enumerate(spike_vector):
         indices = np.arange(spikes.size, dtype=np.int64)
+        if absolut_index:
```
Comment (zm711) [inline src/spikeinterface/core/sorting_tools.py:95]:
```suggestion
        if absolute_index:
```

File: src/spikeinterface/postprocessing/spike_amplitudes.py [hunk 6]
```diff
@@ -127,7 +127,7 @@ def _get_data(self, outputs="numpy"):
         elif outputs == "by_unit":
             unit_ids = self.sorting_analyzer.unit_ids
             spike_vector = self.sorting_analyzer.sorting.to_spike_vector(concatenated=False)
-            spike_indices = spike_vector_to_indices(spike_vector, unit_ids)
+            spike_indices = spike_vector_to_indices(spike_vector, unit_ids, absolut_index=True)
```
Comment (zm711) [inline src/spikeinterface/postprocessing/spike_amplitudes.py:130]:
```suggestion
            spike_indices = spike_vector_to_indices(spike_vector, unit_ids, absolute_index=True)
```

File: src/spikeinterface/postprocessing/spike_locations.py [hunk 7]
```diff
@@ -140,7 +140,7 @@ def _get_data(self, outputs="numpy"):
         elif outputs == "by_unit":
             unit_ids = self.sorting_analyzer.unit_ids
             spike_vector = self.sorting_analyzer.sorting.to_spike_vector(concatenated=False)
-            spike_indices = spike_vector_to_indices(spike_vector, unit_ids)
+            spike_indices = spike_vector_to_indices(spike_vector, unit_ids, absolut_index=True)
```
Comment (zm711) [inline src/spikeinterface/postprocessing/spike_locations.py:143]:
```suggestion
            spike_indices = spike_vector_to_indices(spike_vector, unit_ids, absolute_index=True)
```

**Review by zm711 — COMMENTED** [review]
haha. I think this is all the absolut->absolute for you....

File: src/spikeinterface/core/sorting_tools.py [hunk 8]
```diff
@@ -61,6 +61,12 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
         List of spike vectors optained with sorting.to_spike_vector(concatenated=False)
     unit_ids: np.array
         Unit ids
+    absolute_index: bool, default False
+        Give spike indices absolut usefull when having a unique spike vector
+        or relative to segment usefull with a list of spike vectors
+        When a unique spike vectors (or amplitudes) is used then absolute_index should be True.
+        When a list of spikes (or amplitudes) is used then absolute_index should be False.
```
Comment (alejoe91) [inline src/spikeinterface/core/sorting_tools.py:68]:
```suggestion
        It True, return absolute spike indices, else spike indices are relative to the segment.
        When a unique spike vector is used,  then absolute_index should be True.
        When a list of spikes per segment is used, then absolute_index should be False.
```

File: src/spikeinterface/core/sorting_tools.py [hunk 9]
```diff
@@ -61,6 +61,11 @@ def spike_vector_to_indices(spike_vector: list[np.array], unit_ids: np.array):
         List of spike vectors optained with sorting.to_spike_vector(concatenated=False)
     unit_ids: np.array
         Unit ids
+    absolute_index: bool, default False
+        It True, return absolute spike indices, else spike indices are relative to the segment.
```
Comment (alejoe91) [inline src/spikeinterface/core/sorting_tools.py:65]:
```suggestion
        It True, return absolute spike indices. If False, spike indices are relative to the segment.
```

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**